add variant PZEM-004T-100A-D-P(V1.0)

rmie opened the pull request:
## Description:
I spent some time today trying to use the wrong module for a board marked PZEM-004T-100A-D-P(V1.0).
Hope this is helpful for others.

## Checklist:
  - [x] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

rmie commented:
First of all, this is a pull request to update esphome documentation, not sure if this is the right place to ask for help.

> idk why - i have same module v1 and work with modbus pzemac but curent always zero as well as power and energy- i connected coil correctly checked few times when real measured current 2-3a still see zero- maybe you know what i do wrong?

From your debug output at the end, it looks like Frequency and Voltage measurement is working
=> this would imply that GPIO pins, etc. settings are correct and communication with the module works.

But
- _AC: V=211.2_: is really low for 230V nominal voltage. I haven't had such low voltage ever (Germany)
- _F=50.0 Hz_: is right on the spot.

Double check that these values are actually changing over time, if not, maybe something is wrong with the
mains connection to your module.

Additionally, these modules come in two versions
- 10A range, with internal shunt
- 100A range, with current transformer
The difference here is how current is measured. I'm not sure what happens if e. g. the 10A version is
used with a current transformer or vice versa.

mhl221135 commented:
> First of all, this is a pull request to update esphome documentation, not sure if this is the right place to ask for help.
> 
> > idk why - i have same module v1 and work with modbus pzemac but curent always zero as well as power and energy- i connected coil correctly checked few times when real measured current 2-3a still see zero- maybe you know what i do wrong?
> 
> From your debug output at the end, it looks like Frequency and Voltage measurement is working => this would imply that GPIO pins, etc. settings are correct and communication with the module works.
> 
> But
> 
> * _AC: V=211.2_: is really low for 230V nominal voltage. I haven't had such low voltage ever (Germany)
> * _F=50.0 Hz_: is right on the spot.
> 
> Double check that these values are actually changing over time, if not, maybe something is wrong with the mains connection to your module.
> 
> Additionally, these modules come in two versions
> 
> * 10A range, with internal shunt
> * 100A range, with current transformer
>   The difference here is how current is measured. I'm not sure what happens if e. g. the 10A version is
>   used with a current transformer or vice versa.

i have found issue- my ct broken and regarding 211v i am from Ukraine we do have 140 sometimes and 280 since each 4 hours electricity from city disapear for next 4 hours

swoboda1337 commented on content/components/sensor/pzem004t.md:
```diff
@@ -21,7 +21,8 @@ ESPHome.
 > The PZEM-004, however, is still working (and selling) with this protocol and does not use modbus.
-> For using the newer V3 variant of this sensor please see {{< docref "pzemac" "pzemac" >}}.
+> For using the newer V3 variant or boards marked with PZEM-004T-100A-D-P(V1.0) of this sensor
+> please see {{< docref "pzemac" "pzemac" >}}.
```
Please replace the short code with an md link. I would do it but since you created the pull request directly from current I am unable too.

swoboda1337 commented on content/components/sensor/pzemac.md:
```diff
@@ -151,6 +151,17 @@ modbus_controller:
+
+### PZEM-004T-100A-D-P(V1.0)
+
+![PZEM-004T-100A-D-P(V1.0)](images/pzem-004t-100a-d-p-v1.0.jpg)
```
Picture doesnt load: 
<img width="1378" height="408" alt="image" src="https://github.com/user-attachments/assets/ad9f1c84-1af7-46ca-aba6-07d8dbebc786" />

swoboda1337 commented on content/components/sensor/pzemac.md:
```diff
@@ -151,6 +151,17 @@ modbus_controller:
+
+![PZEM-004T-100A-D-P(V1.0)](images/pzem-004t-100a-d-p-v1.0.jpg)
+
+- The silkscreen marking implies V1.0 which might mislead to try using {{< docref "pzem004t/" >}}.
```
use md link here as well

rmie committed:
fix review comments
pzem004t.md, add PZEM-004T-100A-D-P(V1.0) variant
pzemac.md, add description for PZEM-004T-100A-D-P(V1.0)

ximex commented:
@rmie pls resolve conflicts

ximex reviewed (approved):
looks good now. @swoboda1337 any other change request?
